features/proxy_config.feature

```diff
@@ -335,8 +344,233 @@ Feature: Proxy configuration
+         \* Autogenerated by ubuntu-advantage-tools
+         \* Do not edit this file directly
+         \*
+         \* To change what ubuntu-advantage-tools sets, run one of the following:
+         \* Substitute "apt_https_proxy" for "apt_http_proxy" as necessary.
+         \*   sudo ua config set apt_http_proxy=<value>
+         \*   sudo ua config unset apt_http_proxy
+         \*/
+        Acquire::http::Proxy::esm.ubuntu.com ".*";
+        Acquire::https::Proxy::esm.ubuntu.com ".*";
+        """
+        When I run `apt update` with sudo
+        And I run `cat /var/log/squid/access.log` `with sudo` on the `proxy` machine
+        Then stdout matches regexp:
+        """
+        CONNECT esm.ubuntu.com:443
+        """
+        Then stdout does not match regexp:
+        """
+        .*GET.*ubuntu.com/ubuntu/dists.*
+        """
+        Then stdout does not match regexp:
+        """
+        .*GET.*archive.ubuntu.com.*
+        """
+        Then stdout does not match regexp:
+        """
+        .*GET.*security.ubuntu.com.*
+        """
+        When I run `ua config unset apt_http_proxy` with sudo
+        And I run `ua config unset apt_https_proxy` with sudo
+        And I run `ua refresh config` with sudo
+        Then I verify that no files exist matching `/etc/apt/apt.conf.d/90ubuntu-advantage-aptproxy`
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        ua_config:
+            apt_http_proxy: "invalidurl"
+            apt_https_proxy: "invalidurls"
+        """
+        And I verify that running `ua refresh config` `with sudo` exits `1`
+        Then stderr matches regexp:
+        """
+        "invalidurl" is not a valid url. Not setting as proxy.
+        """
+        When I verify that running `ua config set http_proxy=http://host:port` `with sudo` exits `1`
+        Then stderr matches regexp:
+        """
+        "http://host:port" is not a valid url. Not setting as proxy
+        """
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        ua_config:
+            apt_https_proxy: "https://localhost:12345"
+        """
+        And I verify that running `ua refresh config` `with sudo` exits `1`
+        Then stderr matches regexp:
+        """
+        "https://localhost:12345" is not working. Not setting as proxy.
+        """
+        When I run `ua config set apt_http_proxy=http://<ci-proxy-ip>:3128` with sudo
+        And I run `ua config set apt_https_proxy=http://<ci-proxy-ip>:3128` with sudo
+        When I run `cat /etc/apt/apt.conf.d/90ubuntu-advantage-aptproxy` with sudo
+        Then stdout matches regexp:
+        """
+        /\*
+         \* Autogenerated by ubuntu-advantage-tools
+         \* Do not edit this file directly
+         \*
+         \* To change what ubuntu-advantage-tools sets, run one of the following:
+         \* Substitute "apt_https_proxy" for "apt_http_proxy" as necessary.
```

Comment:
in all tests involving regex matching, you need to escape the quotes, otherwise it is failing
```suggestion
         \* Substitute \"apt_https_proxy\" for \"apt_http_proxy\" as necessary.
```

Reply:
Interesting, this suggestion makes sense, but the existing tests were passing without escaping the quotes (and I thought I saw this new one pass as well, I'll re-run it)

Reply:
Okay, let me update all regex matching tests.

Reply:
It fails consistently here for some reason :|

Reply:
Yeah, updated it since there is no con to adding the backslash.